addons/github/api.py

```diff
@@ -151,7 +151,32 @@ def revoke_token(self):
             return self.gh3.revoke_authorization(self.access_token)
 
     def check_authorization(self):
-        return self.gh3.check_authorization(self.access_token)
+        """Check an authorization created by a registered application.
+
+        OAuth applications can use this method to check token validity
+        without hitting normal rate limits because of failed login attempts.
+        If the token is valid, it will return True, otherwise it will return
+        False.
+
+        :returns:
+            True if token is valid, False otherwise
+        :rtype:
+            bool
+        """
+        p = self.gh3.session.params
+        auth = (p.get('client_id'), p.get('client_secret'))
+        if self.access_token and auth:
```

Comment:
Are `p` or `auth` necessary here? They appear unused. Also:
```
In [1]: bool((None, None))
Out[1]: True
```

Reply:
Apparently not, I just had them because they were used in the method this overrides in the client.